### PR TITLE
container-structure-test: init at 1.18.1

### PR DESCRIPTION
**Pull request by rubenhoenle**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by rubenhoenle** [comment]
tested basic functionality with https://github.com/rubenhoenle/container-structure-test-example

File: pkgs/by-name/co/container-structure-test/package.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      aarch64-darwin = "17cfsxwpaki4x6alcphxf04nby2p3v8kahzjzim4y6ajg3g9pm3h";
+      x86_64-darwin = "099vxrqj1q1xm8fbr37pbxd68vp3jn8w7rwpxdpafxrvfhr617dl";
+    }.${stdenv.hostPlatform.system} or (throw "Unsupported system: ${stdenv.hostPlatform.system}");
+  };
+
+  phases = ["installPhase"];
+
+  installPhase = ''
+    runHook preInstall
+    install -Dm755 $src $out/bin/contaner-structure-test
```
Comment (Noodlez1232) [inline pkgs/by-name/co/container-structure-test/package.nix:28]: typo: should be container

**Review by Noodlez1232 — CHANGES_REQUESTED** [review]

File: pkgs/by-name/co/container-structure-test/package.nix (hunk 1 above)
Comment (rubenhoenle) [reply pkgs/by-name/co/container-structure-test/package.nix]: thanks, done

**Review by Noodlez1232 — APPROVED** [review]
Tested, and it looks good to me!

**Comment by 0x4A6F** [comment]
Result of `nixpkgs-review pr 316449` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>container-structure-test</li>
  </ul>
</details>

File: pkgs/by-name/co/container-structure-test/package.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    homepage = "https://github.com/GoogleContainerTools/container-structure-test";
+    description = "The Container Structure Tests provide a powerful framework to validate the structure of a container image.";
+    license = lib.licenses.asl20;
+    maintainers = with lib.maintainers; [ rubenhoenle ];
+    platforms = [
+      "x86_64-linux"
+      "aarch64-linux"
+      "x86_64-darwin"
+      "aarch64-darwin"
+    ];
```
Comment (Golo300) [inline pkgs/by-name/co/container-structure-test/package.nix:51]:
```suggestion
    platforms = builtins.attrNames sources;
```

**Review by Golo300 — CHANGES_REQUESTED** [review]

**Review by Golo300 — APPROVED** [review]

**Review by 0x4A6F — APPROVED** [review]

File: pkgs/by-name/co/container-structure-test/package.nix [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    runHook postInstall
+  '';
+
+  meta = {
+    homepage = "https://github.com/GoogleContainerTools/container-structure-test";
+    description = "The Container Structure Tests provide a powerful framework to validate the structure of a container image.";
+    license = lib.licenses.asl20;
+    maintainers = with lib.maintainers; [ rubenhoenle ];
+    platforms = builtins.attrNames sources;
+    mainProgram = "container-structure-test";
```
Comment (Aleksanaa) [inline pkgs/by-name/co/container-structure-test/package.nix:47]:
```suggestion
    mainProgram = "container-structure-test";
    sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
```

Comment (Aleksanaa) [reply pkgs/by-name/co/container-structure-test/package.nix:47]: (Not able to build from source?)

Comment (0x4A6F) [reply pkgs/by-name/co/container-structure-test/package.nix:47]: Can be done in a follow-up PR.

**Pull request merged**